kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java

```diff
@@ -89,8 +89,12 @@ public static void checkpoint(Engine engine, Clock clock, SnapshotImpl snapshot)
 
       // Get the metadata of the checkpoint file
       numberOfAddFiles = checkpointDataIter.getNumberOfAddActions();
-    } catch (FileAlreadyExistsException faee) {
-      throw new CheckpointAlreadyExistsException(version);
+    } catch (IOException e) {
```

Comment:
Seems like the hadoop upgrade caused a change in exception type with `FileAlreadyExistsException` instead being the cause